Sources/Ticketmaster/TicketmasterManager.swift

```diff
@@ -55,7 +57,89 @@ class TicketmasterManager {
+            NotificationCenter.default.addObserver(
+                self,
+                selector: #selector(self.receiveTicketmasterNotification),
+                name: NSNotification.Name(rawValue: notificationName),
+                object: nil
+            )
+        }
     }
+    
+    @objc
+    func receiveTicketmasterNotification(_ notification: Foundation.Notification) {
+        guard let roverScreenName = TicketmasterManager.tmEvents[notification.name.rawValue] else {
+            os_log("TicketmasterManager received an unexpected NSNotification, ignoring.", log: .general, type: .error)
+            return
+        }
+        
+        // the same fields are common amongst all the events we monitor for.
+        let attributes: Attributes = [:]
+        
+        let eventAttributes: Attributes = [:]
+        let venueAttributes: Attributes = [:]
+        let artistAttributes: Attributes = [:]
+        
+        if let eventID = notification.userInfo?["event_id"] {
+            eventAttributes["id"] = eventID
+        }
+        
+        if let eventName = notification.userInfo?["event_name"] {
+            eventAttributes["name"] = eventName
+        }
+
+        if let eventDate = notification.userInfo?["event_date"] as? Date {
+            // In order to be (somewhat) consistent with the Android version of the Presence SDK (which yields a pre-rendered date string in a non-standard format), render it into the following format:
+            // Mon, Apr 13, 7 PM
+            let formatter = DateFormatter()
+            formatter.locale = Locale(identifier: "en_US_POSIX")
+            formatter.setLocalizedDateFormatFromTemplate("E, MMM d, h:mm a")
+            eventAttributes["date"] = formatter.string(from: eventDate)
+        }
+        
+        if let eventImageURL = notification.userInfo?["event_image_url"] {
+            eventAttributes["imageURL"] = eventImageURL
+        }
+        
+        if let venueName = notification.userInfo?["venue_name"] {
+            venueAttributes["name"] = venueName
+        }
+        
+        if let venueID = notification.userInfo?["venue_id"] {
+            venueAttributes["id"] = venueID
+        }
+        
+        if let currentTicketCount = notification.userInfo?["current_ticket_count"] {
+            attributes["currentTicketCount"] = currentTicketCount
+        }
+        
+        if let artistName = notification.userInfo?["artist_name"] ?? notification.userInfo?["atrist_name"] /* [sic] */ {
+            artistAttributes["name"] = artistName
+        }
+        
+        if let artistID = notification.userInfo?["artist_id"] {
+            artistAttributes["id"] = artistID
+        }
+        
+        attributes["event"] = eventAttributes
+        attributes["venue"] = venueAttributes
+        attributes["artist"] = artistAttributes
+        
+        let eventInfo = EventInfo(forViewingScreenWithName: roverScreenName, namespace: "ticketmaster", attributes: attributes)
```

Comment:
@chrisrecalis should we pass the event name and event ID as the `contentName` and `contentID` properties here or are you going to do that server-side? If we do it here, maybe you don't have to treat ticketmaster screen views any differently?

Reply:
ah yeah just about to push a commit for that

Reply:
as discussed with @chrisrecalis , not gonna do this one